provider/etcd.go

```diff
@@ -8,12 +10,16 @@ import (
 
 // Etcd holds configurations of the Etcd provider.
 type Etcd struct {
-	Kv `mapstructure:",squash"`
+	Kv  `mapstructure:",squash"`
+	TLS *KvTLS
```

Comment:
I tried that, than it results in 

```
./cmd.go:63: unknown provider.Consul field 'TLS' in struct literal
./cmd.go:67: unknown provider.Etcd field 'TLS' in struct literal
```

So currently I'm trying different ways if I can find something that it simply works. For docker it's much easier to integrate because you've got a single struct of truth, here we have 2 or 3.